minetest/minetest
# Allow exporting config and debug.txt with one click (on all platforms)

For fixing bugs reported by users, you often need their `minetest.conf` and `debug.txt`. However, it can be difficult for users to obtain these files. Desktop users have to find out the location of the Minetest user data directory on their platform. Android users can export `debug.txt` with a button in the app, but getting `minetest.conf` requires a computer with ADB installed if you have a recent Android version.

This PR adds a button for exporting so-called "debug info" to the about tab. This button exists on all platforms. On desktop, it copies to the clipboard, on Android, it opens a share pop-up.

This is how the new about tab looks on desktop:
![new about tab on desktop](https://github.com/minetest/minetest/assets/82708541/44c47ab4-9f42-4fb6-9b55-62938737832c)

This is how the new about tab looks on Android:
![new about tab on android](https://github.com/minetest/minetest/assets/82708541/a6bb2ab6-00ee-409a-8dc0-3207d362877a)

The so-called "debug info" consists of Minetest version, platform, active renderer, `minetest.conf` and `debug.txt`. Here's an example:

```
Minetest 5.8.0-dev-debug-1890e2c3-dirty
Platform: Android
Active renderer: OpenGL ES 3.2 V@0530.0 (GIT@8382da08aa, I139f36ad46, 1645197494) (Date:02/18/22)

minetest.conf
-------------
screen_w = 2412
font_size = 12
enable_3d_clouds = true
gui_scaling = 1.125
screen_h = 1080
maintab_LAST = about
menu_last_game = minetest

debug.txt
---------


-------------
  Separator
-------------

2023-06-18 19:36:57: ERROR[Main]: Game specified in default_game [minetest] is invalid.


-------------
  Separator
-------------

2023-06-18 19:45:54: ERROR[Main]: Game specified in default_game [minetest] is invalid.
```

Inspired by the discussion starting at https://irc.minetest.net/minetest-dev/2023-06-14#i_6091167. Builds on top of https://github.com/minetest/minetest/pull/12370.

## To do

This PR is a Ready for Review.

## How to test

Look at the about tab, verify that it looks as shown in the screenshots. Press the "Copy debug info" / "Share debug info" button and verify that the output is useful for bug reports.

## Q&A

-   Where did the active renderer text go?

    I had to remove it to make space for the new button. However, I think that the active renderer isn't relevant for the average user anyway. If you really need this information, you can still get it via the "Copy debug info" button.

-   Why does the "minetest.net" button now use the full available width?

    Because the text on this button was cut off on my Android device.
